### PR TITLE
Add find field structure and find messages structure APIs

### PR DESCRIPTION
**Pull request by lcawl**
This PR adds specifications based on the following documentation:

- https://www.elastic.co/guide/en/elasticsearch/reference/master/find-field-structure.html
- https://www.elastic.co/guide/en/elasticsearch/reference/master/find-message-structure.html

### CODE REVIEW

File: specification/text_structure/find_field_structure/FindFieldStructureRequest.ts [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+    ecs_compatibility?: EcsCompatibilityType
+    /**
+     * If true, the response includes a field named `explanation`, which is an array of strings that indicate how the structure finder produced its result.
+     * @server_default false
+     */
+    explain?: boolean
+    /**
+     * The field that should be analyzed.
+     */
+    field: string
```
Comment (flobernd) [inline specification/text_structure/find_field_structure/FindFieldStructureRequest.ts:70]:
```suggestion
    field: Field
```

File: specification/text_structure/find_field_structure/FindFieldStructureRequest.ts [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+     * If the format is `semi_structured_text`, you can specify a Grok pattern that is used to extract fields from every message in the text.
+     * The name of the timestamp field in the Grok pattern must match what is specified in the `timestamp_field` parameter.
+     * If that parameter is not specified, the name of the timestamp field in the Grok pattern must match "timestamp".
+     * If `grok_pattern` is not specified, the structure finder creates a Grok pattern.
+     */
+    grok_pattern?: GrokPattern
+    /**
+     * The name of the index that contains the analyzed field.
+     */
+    index: string
```
Comment (flobernd) [inline specification/text_structure/find_field_structure/FindFieldStructureRequest.ts:88]:
```suggestion
    index: IndexName
```

File: specification/text_structure/find_field_structure/FindFieldStructureRequest.ts [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+     *
+     * If the format is `semi_structured_text`, this field must match the name of the appropriate extraction in the `grok_pattern`.
+     * Therefore, for semi-structured text, it is best not to specify this parameter unless `grok_pattern` is also specified.
+     *
+     * For structured text, if you specify this parameter, the field must exist within the text.
+     *
+     * If this parameter is not specified, the structure finder makes a decision about which field (if any) is the primary timestamp field.
+     * For structured text, it is not compulsory to have a timestamp in the text.
+     */
+    timestamp_field?: string
```
Comment (flobernd) [inline specification/text_structure/find_field_structure/FindFieldStructureRequest.ts:120]:
```suggestion
    timestamp_field?: Field
```

File: specification/text_structure/find_field_structure/FindFieldStructureResponse.ts [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+export class Response {
+  body: {}
```
Comment (flobernd) [inline specification/text_structure/find_field_structure/FindFieldStructureResponse.ts:20]: This is TBD?

**Review by flobernd — COMMENTED** [review]
Looking good 🙂 Left a few type related suggestions.

File: specification/text_structure/find_field_structure/FindFieldStructureResponse.ts (hunk 4 above)
Comment (lcawl) [reply specification/text_structure/find_field_structure/FindFieldStructureResponse.ts]: I've added the values that appear in the docs but I investigated further into whether the endpoints all actually return the same set.

Comment (flobernd) [reply specification/text_structure/find_field_structure/FindFieldStructureResponse.ts]: Thank you 🙂 We can always improve in a later iteration, if not everything is mapped 100% correctly.

**Review by flobernd — APPROVED** [review]

**Pull request merged**